packages/web-components/src/rating-display/rating-display.ts

```diff
@@ -0,0 +1,122 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { RatingDisplayColor, RatingDisplaySize } from './rating-display.options.js';
+
+/**
+ * The base class used for constructing a fluent-rating-display custom element
+ * @public
+ */
+export class RatingDisplay extends FASTElement {
+  /**
+   * The internal {@link https://developer.mozilla.org/docs/Web/API/ElementInternals | `ElementInternals`} instance for the component.
+   *
+   * @internal
+   */
+  public elementInternals: ElementInternals = this.attachInternals();
+
+  /**
+   * The color of the rating display icons.
+   *
+   * @public
+   * @default `marigold`
+   * @remarks
+   * HTML Attribute: `color`
+   */
+  @attr
+  public color: RatingDisplayColor = RatingDisplayColor.marigold;
+
+  /**
+   * Renders a single filled icon with a label next to it.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `compact`
+   */
+  @attr({ attribute: 'compact', mode: 'boolean' })
+  public compact: boolean = false;
+  protected compactChanged(): void {
+    if (this.compact) {
+      this.max = 1;
+    }
```

Comment:
This would be an odd side effect to me, where an attribute I set is _reset_ from within the class. Can we use a getter instead to guard against the delta here?

Reply:
I've made the change to check for `compact` right where we need to do a calculation using `max`. That way, we don't need to have this side effect or a getter. Thanks for calling it out.